### ISSUE TITLE
Old setting mechanism detection broken in maven 3.6.2

### ISSUE THREAD

**Issue by aroemers** [jgitver/jgitver-maven-plugin#119]
## Issue

**version**: 1.5.1

**usage context**:

- [x] maven command line: 3.6.1 / 3.6.2 

**Problem description**:

The "old setting mechanism" detection seems to be broken with Maven 3.6.2. 
It gives a false positive, failing the build:

```
Using Docker executor with image maven:3.6-jdk-11
...
WARNING] jgitver has changed!
[WARNING] 
[WARNING] it now requires the usage of maven core extensions (> 3.3.1) instead of standard plugin extensions.
[WARNING] The plugin must be now declared in a `.mvn/extensions.xml` file.
[WARNING] 
[WARNING]     read https://github.com/jgitver/jgitver-maven-plugin for further information
[WARNING] 
[ERROR] detection of jgitver old setting mechanism: jgitver must now use maven core extensions only -> [Help 1]
```

The detection is fine with Maven 3.6.1, using Docker image `maven:3.6.1-jdk-11`.

**Comment by McFoggy** [comment]
other things are broken in regards of core extension : see [mailing-list](https://mail-archives.apache.org/mod_mbox/maven-users/201909.mbox/%3Cql5i4k%247fuk%241%40blaine.gmane.org%3E)

**Comment by soloturn** [comment]
am affected as well using java8. but i am not too sure what changed in maven which makes it react like this. in JGitverExtension.java there is 
```
if (JGitverModelProcessor.class.isAssignableFrom(modelProcessor.getClass())) {
    if (!mavenSession.getUserProperties().containsKey(JGitverUtils.SESSION_MAVEN_PROPERTIES_KEY)) {
        JGitverUtils.failAsOldMechanism(c);
    }
} else {
    JGitverUtils.failAsOldMechanism(c);
}
```
does it make sense to still have this detection?

**Comment by McFoggy** [comment]
I think maven maintainers broke in `3.6.2` the usage of `@Component(role = ModelProcessor.class)`.

At least `JGitverModelProcessor`is not called anymore in maven-3.6.2.

I'll raise the point on the mailing list.

**Comment by qoomon** [comment]
I think this commit causing the problems https://github.com/apache/maven/commit/353d3eaa8f2be47878f584a7da398572b41d90af

**Comment by archiecobbs** [comment]
I'm having this same problem.

For me the bug happens with this version on Linux:

> Apache Maven 3.6.2 (SUSE 3.6.2-lp150.13.7)

but not with this version on Mac OS X Homebrew:

> Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T14:00:29-05:00)

This workaround seems to work: add `-Djgitver.session=true` to the `mvn` command line.

(What is `gitver.session` and why is it required? Why not just remove that line?)

**UPDATED TO ADD:**

The workaround mentioned above does not actually work: it does allow the build to proceed, but the version is not updated.

**Comment by soloturn** [comment]
@McFoggy,  @archiecobbs, maven-3.6.3 works for me with jgitver-maven-plugin, see https://github.com/apache/maven/commit/16961b32b7dccbc9447ed4e18b4e0949d5fe3b56 .

**Comment by McFoggy** [comment]
https://issues.apache.org/jira/browse/MNG-6765 has corrected the regression in 3.6.2

**Comment by soloturn** [comment]
this is the commit:
https://github.com/apache/maven/commit/07ab962c85950b034be3216996900920c0204c3a
